### DISCUSSION TITLE
React ou Vue?

### DISCUSSION THREAD

**Discussion by gabrielgrs** [frontendbr/forum#785] [Perguntas e Respostas]
Boa tarde pessoal, tudo bem?

Seguinte, estou trabalhando com NodeJS para poder criar aplicações pessoais e pequenos projetos rapidamente, sem perder tanto tempo com trabalhos desnecessários, já terminei o Back-end de uma aplicação e estou partindo para o Front-End! Na faculdade estou tendo aula com uso de Angular 4, gostei dele, mas talvez por ser um Framework, em minha questão de querer um processo ágil, ele não está me respondendo bem. Então comecei a olhar Vue e React!

Usei um pouco de cada, mas percebi que o Vue, por mais simples que seja, tem comportamentos estranhos com o compilador, com por exemplo ter uma linha em branco entre um { }, ele apontar erro. Coisas banais. Mas o React está me parecendo não ser a solução ideal para "consumir uma API e por em produção logo!".

Então, alguém poderia me auxiliar com qual escolher, considerando que irei utilizar apenas em projetos de pequeno ou médio porte?

Obrigado!

**Comment by ghost** [comment]
Sobre os projetos de pequeno porte @gabrielgrs eu não sei qual a complexidade quando você fala em pequeno porte, mas consideraria fazer apenas com JS moderno.

Médio porte e respondendo você sobre o uso de React ou Vuejs, depende de inúmeras coisas.
`comportamentos estranhos com o compilador` eu realmente não sei o que isso pode ser você poderia criar alguma issue pra saber se alguém que usa vuejs não tem a solução para teus problemas. O VueJS tem uma curva de aprendizado muito rápida então talvez ele seja uma boa opção para teu contexto.

`Mas o React está me parecendo não ser a solução ideal para "consumir uma API e por em produção logo!".`

Bom não sei se você falou isso pela curva de aprendizado e o tanto que tu terás que aprender, se você quer aprender a tecnologia vai fundo mesmo que "você ache que vai demorar".

Mas antes de considerar qualquer aprendizado certifique-se que você consegue usar bem javascript e mais do que isso resolve teu problema, depois melhora a tua solução. E isso pode ser usar uma lib para resolver datas, resolver problemas de performance.

Espero ter te ajudado!

**Comment by AlbericoD** [comment]
sem frameworks mesmo,  é mais rápido(supondo que vc domine bem e talz), o bom e velho javascript, react trabalha com chamadas em arvores, não parece ser o que você busca.

**Comment by klarkc** [comment]
Já usei ambos (mais o Vue), mas vou com o @AlbericoD se vc quer entregar logo, cara vai no que conhece, seja VanillaJS ou qualquer outro framework :p agora se vc quer se aventurar num framework novo vale a pena ler a documentação de ambos, principalmente as comparações, então você poderá decidir por si mesmo qual é melhor, pois cada um tem vai ter uma opinião pendente pro que mais domina.

**Comment by gjrmacedo** [comment]
Dada as opções que você deu, vou de Vue fácil, curva de aprendizado pequena e entrega tão bem quanto o React o faz. Sobre os comportamentos estranhos que você comentou, é pura configuração do Webpack, nada que deveria te impedir.
Essa é minha opinião pessoal, dada a experiencia que tive com as duas libs.

**Comment by caioincau** [comment]
O que você está chamando de "Comportamentos estranhos", nada mais é do que o linter, que inclusive é de boa prática se usar no node tbm.

Recomendo ver sobre "como configurar eslint", para deixar ele mais próximo do seu modo de escrever código, mas se você está usando vue-cli, provavelmente você pode remover o eslint na hora de dar start no projeto.

**Comment by maclovin** [comment]
Para projetos pequenos recomendo Vue ou Gatsby. Com o CLI deles você começa um projeto baseado em um template muito rápido.

**Comment by AmandaFonseca** [comment]
Fugindo totalmente do topico @maclovin me diga que o seu nome foi inspirado no McLovin do filme Superbad por favor kkkkk 
Eu amo aquele as cenas dele no filme.

**Comment by munizart** [comment]
@gabrielgrs pessoalmente achei vue muito mais facil de aprender...
Da pra usar ele com JSX tambem, resolveria o problema das linhas em branco entre as chaves.

Uma serie legal pra pegar é essa: https://www.youtube.com/watch?v=Fa4cRMaTDUI
Bons estudos!

**Comment by ROBERTOCONRADO** [comment]
Hoje o React

**Comment by ojvribeiro** [comment]
UI Frameworks são praticamente obrigatórios hoje em dia. Eu diria pra você focar na base (HTML, CSS e JS) primeiro, depois escolher o framework que você gostar mais. Algumas pessoas vão dizer pra você ir direto pro React porque tem mais vagas, mas eu reforço: vá naquele que você se identificar mais, porque todas as comunidades além do React são boas o bastante pra ingressar. Depois que você pegar o modo de pensar do framework (em relação ao vanilla), todos os outros vão ficar extremamente mais fáceis de aprender.

Me tirando como exemplo, eu uso Vue, tanto no trabalho (empresa gringa que todo o front é formado por BRs) quanto em side-projects (uso Vue + [Nuxt](https://nuxt.com) pra elas). Eu me identifico com o Vue, gosto de desenvolver com Vue. Mas há quem goste do React, e tá tudo bem. Só não caia nesse papo de mercado e ecossistema, porque isso só faz criar uma falsa impressão de outras techs que não são o React. Se você for dev React, você vai conseguir vagas em React, se for dev Vue, vai conseguir vaga pra Vue, e assim por diante.

Resumindo tudo: estude a tríade HTML, CSS e JS primeiro. Depois escolha um framework que você se sentir mais confortável. Os mais populares e maduros hoje, pra facilitar sua escolha, são:

- React / Next.js
- Vue / Nuxt
- Svelte / SvelteKit
- Angular

Existem muitas outras alternativas, e você pode escolher qualquer uma delas, que a migração pra próxima vai ser moleza.

**Comment by nazarepiedady** [comment]
Quem estiver interessado em aprender ou estudar a versão atual da Vue, pode consultar este material: [vuejs-translations/docs-pt](https://github.com/vuejs-translations/docs-pt)
